src/infrastructure/repositories/ViemChainMetricsRepository.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  createPublicClient,
+  http,
+  PublicClient
+} from "viem"
+// @ts-ignore
+import { hemiSepolia, hemi } from "hemi-viem"
+// @ts-ignore
+import { Hash } from '../../domain/valueObjects/Hash'
+import { ChainMetricsRepository } from '../../domain/repositories/ChainMetricsRepository'
+import { ChainMetrics } from '../../domain/entities/ChainMetrics'
+import { BlockNumber } from '../../domain/valueObjects/BlockNumber'
+import { Metric } from '../../domain/valueObjects/Metric'
+
+export class ViemChainMetricsRepository implements ChainMetricsRepository {
+  private client: PublicClient
+  private blockExplorerUrl: string
+
+  constructor() {
+    const chain = !!process.env['TESTNET'] ? hemiSepolia : hemi
+    
+    // @ts-expect-error
+    this.client = createPublicClient({ chain, transport: http() })
+    this.blockExplorerUrl = process.env['BLOCK_EXPLORER_URL'] || ''
```

Comment:
Do we need an env var for block explorer URL here? `Hemi-viem` already exposes it, for example: `hemiSepolia.rpcUrls.default.http[0]`